HLTrigger/Configuration/python/HLT_75e33/modules/hltTriggerSummaryAOD_cfi.py

```diff
@@ -2,7 +2,7 @@
 
 hltTriggerSummaryAOD = cms.EDProducer( "TriggerSummaryProducerAOD",
     throw = cms.bool( False ),
-    processName = cms.string( "@" ),
-    moduleLabelPatternsToMatch = cms.vstring( 'hlt*' ),
+    processName = cms.string( "HLT" ),
```

Comment:
So, this is the problematic thing: the L1T objects are being made in a different process now. If we use `processName = cms.string( "@currentProcess" )`, this module will complain endlessly about "Uunknown pid". Should we really do that?

Reply:
Mh.. then I'm missing something; I thought `@currentProcess` here would be equivalent to using `HLT`. I'll have to recheck, you can ignore this unless I come back to this point.

Reply:
Well, as far as I know the relval workflows for 75e33 phase2 HLT run HLT:fake2 together with L1T in step2 with process name HLT, while the actual phase2 HLT is run in step4 with process name HLTX. So it looks like the above will pack up L1T and HLT:fake2 collections which does not seem to be correct for HLT as fake2 is not the target, but 75e33 is... Have you actually checked whether what is packed up is correct? What lead you to change away from the original "@"? Could you check whether processName = "*" would work and work properly?

Reply:
OTOH you also change elsewhere:
```
L1TkMuons = cms.EDProducer("L1TkMuonProducer",
```
so you run some L1T collection producers within HLTX, so should the process name be "HLTX" instead of "HLT"?
Or do you need L1T collections from both process HLT and process HLTX?

Reply:
Hi Martin!
 - I will leave the "check the correctness of the packing" to the L1T team, this is the PR for their MC production. When this is in, I would say that the next step is get rid of this "fake step2 + real step4" thing and go back to a more sane modus operandi. But I don't want to block their PR now.
 - What lead me to change away from the original "@" were some "Uunkown pid" errors related to L1T collections. I gave it a try on debugging, but couldn't. And again, since this is for the L1T team, I don't think it is much fruitful for us to debug it at this level (but I may be wrong!)
 -  processName = "*" just gives me a lot of "Uunknown pid" again :(
 - As far as I know, all L1T collections should be from process HLT now.

Reply:
Hmm, I do not think L1T will check HLT code. They will not be affected as this packing up is for HLT usage & HLT benefit.
Why is the 'sane' modus not possible right now?

Reply:
>  Why is the 'sane' modus not possible right now?

This hasn't been answered. Also, for my education: does this Phase-2 MC production follow exactly what is done in wf `39434.75` (or some other sequence of steps)?

Anyway, I tried to understand why `"*"` wouldn't work, and I think there might be a bug in `TriggerSummaryProducerAOD`:
```diff
diff --git a/HLTrigger/HLTcore/plugins/TriggerSummaryProducerAOD.cc b/HLTrigger/HLTcore/plugins/TriggerSummaryProducerAOD.cc
index bac8eb0f3a3..53f83a65c95 100644
--- a/HLTrigger/HLTcore/plugins/TriggerSummaryProducerAOD.cc
+++ b/HLTrigger/HLTcore/plugins/TriggerSummaryProducerAOD.cc
@@ -262,8 +262,8 @@ void TriggerSummaryProducerAOD::produce(edm::StreamID, edm::Event& iEvent, const
   /// Record the InputTags of those L3 filters and L3 collections
   std::vector<bool> maskFilters;
   maskFilters.resize(nfob);
-  InputTagSet filterTagsEvent(pn_ != "*");
-  InputTagSet collectionTagsEvent(pn_ != "*");
+  InputTagSet filterTagsEvent(pn_ == "*");
+  InputTagSet collectionTagsEvent(pn_ == "*");
 
   unsigned int nf(0);
   for (unsigned int ifob = 0; ifob != nfob; ++ifob) {
```
(this sets up the flag `ignoreProcess`, which should be `true` for `pn_ == "*"`)

If I use that patch, and the following module (which runs in the Phase-2 HLT in step4)
```py
process.hltTriggerSummaryAOD = cms.EDProducer("TriggerSummaryProducerAOD",
    moduleLabelPatternsToMatch = cms.vstring(
        'hlt*',
        'L1Tk*',
        'l1ct*',
        '*L1T*',
    ),
    moduleLabelPatternsToSkip = cms.vstring(),
    processName = cms.string('*'),
    throw = cms.bool(False)
)                                                                                                                                                                                                  
```
I don't get any complaints about unknown collections, and I see trigger-objects with L1T and HLT ids in the output file.

If the issue with `"*"` in `TriggerSummaryProducerAOD` is confirmed, I'll make a separate PR for that. And if so, `hltTriggerSummaryAOD` should likely be fixed in this PR.

Reply:
Hi Thiago, I think the point is that `hltTriggerSummaryAOD` isn't giving the correct outputs in wf `39434.75`. To fix that, as far I know now, one would have to (1) apply a fix to `TriggerSummaryProducerAOD`, and (2) update the module `hltTriggerSummaryAOD` in the Phase-2 HLT to be
```
process.hltTriggerSummaryAOD = cms.EDProducer("TriggerSummaryProducerAOD",
    moduleLabelPatternsToMatch = cms.vstring(
        'hlt*',
        'L1Tk*',
        'l1ct*',
        '*L1T*',
    ),
    moduleLabelPatternsToSkip = cms.vstring(),
    processName = cms.string('*'),
    throw = cms.bool(False)
)
```
I can take care of (1). Regarding (2), if you think this can be done in a follow-up PR, that's fine by me; otherwise it should be done here.

Then there were a couple of follow-up questions to https://github.com/cms-sw/cmssw/pull/38442#discussion_r949321463.

 - As Martin alluded to, anything related to `TriggerSummaryProducerAOD` needs to be fixed by HLT, it won't be done by L1T.
 - The plan/timeline to "get rid of this fake step2 + real step4 thing and go back to a more sane modus operandi" is unclear. What should we expect here, and when?
 - Finally, will this Phase-2 MC production follow exactly what is done in wf `39434.75`, or does it use a different sequence of steps?

Reply:
Allow me to jump in.
For this:
- Finally, will this Phase-2 MC production follow exactly what is done in wf 39434.75, or does it use a different sequence of steps?
The production is mainly for L1T, so we can stop at RECO step. If HLT needs to use this production also, we can follow .75 workflow. Or you can rerun it on top privately as 12_3 campaign.

- The plan/timeline to "get rid of this fake step2 + real step4 thing and go back to a more sane modus operandi" is unclear. What should we expect here, and when?
I think it is possible to make it works as I try last time (discussed it in one of upgrade HLT meeting). But this should be done and be validated. Maybe it can be target for 12_6 if @trtomei agrees.

Reply:
On a related note, why do we use `@currentProcess` anywhere in the HLT configurations ?
IIRC, it should be needed only rarely, to break some kind of wrong or circular dependency among modules...

Reply:
"@" predates the introduction of @currentProcess by the Fwk. It was/is used to denote figuring out the name of the current process and use it, as usually the summary producer is part of HLT and run in the same current process (but also having the possibility to use an explicit process name)

Reply:
The need for 
```
moduleLabelPatternsToMatch = cms.vstring(
        'hlt*',
        'L1Tk*',
        'l1ct*',
        '*L1T*',
    ),
```
really points to the requirement of a cleanup of the collections and their labelling. It is also prone to false positives in wildcard matching.

Reply:
Does '*L1T*' include 'L1Tk*' ?

Reply:
> really points to the requirement of a cleanup of the collections and their labelling.

Wholeheartedly agree... I'm not sure what `l1ct` stands for (L1 Calo Trigger?) but it would be really helpful if all L1 trigger modules where named `l1tSomething`.

Reply:
> On a related note, why do we use `@currentProcess` anywhere in the HLT configurations ?

In the [current Run-3 GRun](https://raw.githubusercontent.com/cms-sw/cmssw/0139c1f048bab9beda90bc8083357c366152e5d3/HLTrigger/Configuration/test/OnLine_HLT_GRun.py) (I'm digressing from Phase 2) we have 27 occurrences of `@currentProcess`; they are mostly in Scouting modules, and I don't know the full story, but I remember from [here](https://its.cern.ch/jira/browse/CMSHLT-2150?focusedCommentId=4024471&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-4024471) that this was done when the scouting packers were still on EndPaths, to ensure only collections from the current process were packed, never falling back on collections in the input EDM files (since, iiuc, for MC the input files can already contain HLT-scouting collections with those same names). Now that the packers run inside the Scouting `Path`s, this shouldn't be an issue anymore, and indeed I think `@currentProcess` could be removed.

There are also a couple of instances of `TriggerResultsFilter` that use `usePathStatus=True` but also specify
```py
hltResults = cms.InputTag( 'TriggerResults','','@currentProcess' ),
```
and indeed the latter InputTag should emptied, so also here no need for `@currentProcess`.

Reply:
> "@" predates the introduction of @currentProcess

After looking into `TriggerSummaryProducerAOD`, it seems to me that doesn't support `processName = cms.string( "@currentProcess" )` (all the InputTag comparisons won't work if the process name is literally "@currentProcess"), so https://github.com/cms-sw/cmssw/pull/38442#discussion_r948858310 was a wrong suggestion.

Maybe `TriggerSummaryProducerAOD` should be updated to either treat `@currentProcess` like it treats `@`, or at least give a warning if `@[something]` is used?

Reply:
@currentProcess is only valid as the process name within an InputTag. If someone specifies an non-existing process name for the string in this plugin, nothing will match, that's the expected behaviour...

Reply:
The point is that that may not be obvious to a user. Anyway, fine, it was a minor point.